frb_codegen/src/generator/rust/ty_general_list.rs

```diff
@@ -17,22 +17,36 @@ impl TypeGeneralListGenerator<'_> {
 
 impl TypeRustGeneratorTrait for TypeGeneralListGenerator<'_> {
     fn wire2api_body(&self) -> Option<String> {
-        Some(TypeGeneralListGenerator::WIRE2API_BODY.to_string())
+        Some(match &self.ir {
+            IrTypeGeneralList::GeneralList { .. } => TypeGeneralListGenerator::WIRE2API_BODY.into(),
+            IrTypeGeneralList::GeneralArray { len, .. } => format!(
```

Comment:
Similarly, is it possible to delegate GeneralArray to a GeneralList to avoid unsafe operations

Reply:
This part now has a new change, I'm not sure if it fits in more minimally, you can check it out.